crates/forge-runner/src/gas.rs

```diff
@@ -37,11 +39,30 @@ pub fn calculate_used_gas(
         resources.execution_resources,
     );
 
-    let gas = l1_and_vm_costs + messaging_gas_vector;
+    let events_costs = get_events_cost(&resources.events, transaction_context);
+
+    let gas = l1_and_vm_costs + messaging_gas_vector + events_costs;
 
     Ok(gas.l1_gas)
 }
 
+fn get_events_cost(events: &[EventContent], transaction_context: &TransactionContext) -> GasVector {
```

Comment:
We can pass a vector and consume it so we get rid of the clone. Just unpack the struct in the previous function to satisfy the borrow checker
```rs
let UsedResources {events, a, b, c} = used_resources;
```

Reply:
Done